suite/testcommon.py

```diff
@@ -1205,3 +1255,168 @@ def test_univeral_loader(self):
+                assert(ins.operation == binja.MediumLevelILOperation.MLIL_IF)
+                # test if condition value is updated to true
+                assert(ins.condition.value == True)
+                # test if register value is updated to 0
+                assert(ins.get_reg_value_after('r3') == 0)
+                # test if branch is eliminated in hlil
+                for hlil_ins in func.hlil.instructions:
+                    assert(hlil_ins.operation != binja.HighLevelILOperation.HLIL_IF)
+
+                # test undo after round trip
+                bv.undo()
+                bv.update_analysis_and_wait()
+                ins_idx = func.mlil.get_instruction_start(0x845c)
+                ins = func.mlil[ins_idx]
+                assert(ins.operation == binja.MediumLevelILOperation.MLIL_IF)
+                # test if condition value is updated to undetermined
+                assert(ins.condition.value.type == binja.RegisterValueType.UndeterminedValue)
+                # test if register value is updated to undetermined
+                assert(ins.get_reg_value_after('r3').type == binja.RegisterValueType.EntryValue)
+                # test if branch is restored in hlil
+                found = False
+                for hlil_ins in func.hlil.instructions:
+                    if hlil_ins.operation == binja.HighLevelILOperation.HLIL_IF:
+                        found = True
+                assert(found)
+
+                bv.file.close()
+
+            os.unlink(temp_name)
+            # Check retrieval of user variable values
+            return True
+        finally:
+            self.delete_package("helloworld")
+
+    def test_possiblevalueset_ser_and_deser(self):
+        """PossibleValueSet serialization and deserialization"""
+        def test_helper(value):
+            file_name = self.unpackage_file("helloworld")
+            try:
+                with binja.BinaryViewType['ELF'].open(file_name) as bv:
+                    bv.update_analysis_and_wait()
+                    func = bv.get_function_at(0x00008440)
+
+                    ins_idx = func.mlil.get_instruction_start(0x845c)
+                    ins = func.mlil[ins_idx]
+
+                    var = ins.vars_read[0]
+                    defs = func.mlil.get_var_definitions(var)
+                    def_site = defs[0].address
+
+                    func.set_user_var_value(var, def_site, value)
+                    bv.update_analysis_and_wait()
+
+                    def_ins_idx = func.mlil.get_instruction_start(def_site)
+                    def_ins = func.mlil[def_ins_idx]
+
+                    assert(def_ins.get_possible_reg_values_after('r3') == value)
+
+                    temp_name = next(tempfile._get_candidate_names()) + ".bndb"
+                    bv.create_database(temp_name)
+                    bv.file.close()
+
+                with binja.FileMetadata(temp_name).open_existing_database(temp_name).get_view_of_type('ELF') as bv:
```

Comment:
```
with binja.open_view(temp_name) as bv:
```
works here too. ( I realize the unit tests might not follow this convention but this is now preferred)
If you do this then no need for `bv.file.close()` and no need for `update_analysis_and_wait()`